aws/ecs/cluster/main.tf

```
@@ -3,6 +3,19 @@
  */
 resource "aws_ecs_cluster" "ecs_cluster" {
   name = var.cluster_name == "" ? "${var.app_name}-${var.app_env}" : var.cluster_name
+
+  tags = merge({
+    app_name = var.app_name
+    app_env  = var.app_env
+  }, var.tags)
+
+  lifecycle {
+    ignore_changes = [
+      # Ignore changes to tags, e.g. because a management agent
+      # updates these based on some ruleset managed elsewhere.
+      tags["AWS.SSM.AppManager.ECS.Cluster.ARN"],
+    ]
```

Could we use a variable to make this optional?
```suggestion
    ignore_changes = var.lifecycle_ignore_changes    
```

And the variable:
```
variable "lifecycle_ignore_changes" {
  type        = list(string)
  description = "list of resources that should be added to the ignore_changes lifecycle rule"
  default     = []
}

---

It seems ignore_changes doesn't accept variables.  So I've taken that out.

---

Oh, I'm sorry to have misled you. I see there is an `ignore_tags` available on the [aws provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#ignore_tags-configuration-block) that might help.